superset/views/core.py

```diff
@@ -483,6 +492,43 @@ def annotation_json(  # pylint: disable=no-self-use
         payload = viz_obj.get_payload()
         return data_payload_response(*viz_obj.payload_json_and_has_error(payload))
 
+    @event_logger.log_this
+    @api
+    @has_access_api
+    @handle_api_exception
+    @permission_name("explore_json")
+    @expose("/explore_json/data/<cache_key>", methods=["GET"])
```

Comment:
New API endpoint for fetching cached chart data, as well as cached `form_data` in order to do request-time permission checks.